community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/ast/ResolvedCall.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v3_0.ast
+
+import org.neo4j.cypher.internal.compiler.v3_0.spi.{ProcedureSignature, QualifiedProcedureName}
+import org.neo4j.cypher.internal.frontend.v3_0.SemanticCheckResult._
+import org.neo4j.cypher.internal.frontend.v3_0._
+import org.neo4j.cypher.internal.frontend.v3_0.ast.Expression.SemanticContext
+import org.neo4j.cypher.internal.frontend.v3_0.ast._
+import org.neo4j.cypher.internal.frontend.v3_0.symbols.{CypherType, _}
+
+object ResolvedCall {
+  def apply(signatureLookup: QualifiedProcedureName => ProcedureSignature)(unresolved: UnresolvedCall): ResolvedCall = {
+    val UnresolvedCall(_, _, declaredArguments, declaredResults) = unresolved
+    val position = unresolved.position
+    val signature = signatureLookup(QualifiedProcedureName(unresolved))
+    val callArguments = declaredArguments.getOrElse(signatureArguments(signature, position))
+    val callResults = declaredResults.getOrElse(signatureResults(signature, position))
+    ResolvedCall(signature, callArguments, callResults, declaredArguments.nonEmpty, declaredResults.nonEmpty)(position)
+  }
+
+  def signatureArguments(signature: ProcedureSignature, position: InputPosition): Seq[Parameter] = {
+    signature.inputSignature.map { field => Parameter(field.name, CTAny)(position) }
+  }
+
+  def signatureResults(signature: ProcedureSignature, position: InputPosition): Seq[ProcedureResultItem] = {
+    signature.outputSignature.getOrElse(Seq.empty).map { field => ProcedureResultItem(Variable(field.name)(position))(position) }
+  }
+}
+
+case class ResolvedCall(signature: ProcedureSignature,
```

Comment:
I would like the `fakeDeclarations`, `coerceArguments` and `returnColumns` to have explicitly declared types.